packages/synchro-charts/src/components/sc-dial/sc-dial-base/sc-dial-base.tsx

```diff
@@ -0,0 +1,205 @@
+  if (dataStream) {
+    return dataStream.detailedName || dataStream.name;
+  }
+  return null;
+};
+@Component({
+  tag: 'sc-dial-base',
+  styleUrl: 'sc-dial-base.css',
+  shadow: false,
+})
+export class ScDialBase {
+  @Prop() viewport: ViewPortConfig;
+  @Prop() breachedThreshold?: Threshold;
+  @Prop() valueColor?: string; // css color string
+
+  @Prop() alarmStream?: DataStream;
+
+  @Prop() propertyStream?: DataStream;
+  @Prop() propertyPoint?: DataPoint<Primitive>;
+  @Prop() size?: MinimalSizeConfig = { width: 0, height: 0 };
+
+  @Prop() isLoading?: boolean = false;
+
+  @State() fontSize: TextSizeConfig = sizeContent.XL;
+
+  getColorByPercent = (percent: number) => {
+    // eslint-disable-next-line no-nested-ternary
+    return percent < 0.3333
+      ? StatusProgress.CRITICAL
+      : percent < 0.6667
+      ? StatusProgress.WARNING
+      : StatusProgress.NORMAL;
+  };
+
+  @Watch('propertyStream')
+  getSize(newVal: DataStream) {
+    const dataLength = newVal.data.length || 0;
+    const textLength =
+      dataLength && (newVal.data[dataLength - 1].y.toString().length || 0) + (newVal.unit?.length || 1);
+
+    if (textLength >= 0 && textLength < 2) {
+      this.fontSize = sizeContent.Huge;
+    }
+    if (textLength >= 2 && textLength < 7) {
+      this.fontSize = sizeContent.XL;
+    }
+
+    if (textLength >= 7 && textLength < 9) {
+      this.fontSize = sizeContent.L;
+    }
+
+    if (textLength >= 9 && textLength < 12) {
+      this.fontSize = sizeContent.M;
+    }
+
+    if (textLength >= 12 && textLength < 14) {
+      this.fontSize = sizeContent.S;
+    }
+
+    if (textLength >= 14) {
+      this.fontSize = sizeContent.XS;
+    }
+  }
+
+  render() {
+    const icon = this.breachedThreshold ? this.breachedThreshold.icon : undefined;
+    const { yMin = 0, yMax = 0 } = this.viewport;
+    const propertyStream = this.propertyStream ? isNumberDataStream(this.propertyStream) && this.propertyStream : null;
+    const point = propertyStream ? this.propertyPoint : null;
+    const error = propertyStream ? propertyStream.error : 'Only numbers are supported';
+
+    const percent = point ? (point.y as number) / (yMax - yMin) : 0;
+    const labelColor =
+      this.valueColor || (this.breachedThreshold ? this.getColorByPercent(percent) : StatusProgress.BLUE);
+    const left = 760 * (1 - percent);
+    const right = 760 - left;
+    const stream = propertyStream;
+    const unit = propertyStream && propertyStream.unit;
+    const width = `${(this.size?.width || 0) - (this.size?.marginLeft || 0) - (this.size?.marginRight || 0)}px`;
+    const height = `${(this.size?.height || 0) - (this.size?.marginTop || 0) - (this.size?.marginBottom || 0) + 80}px`;
+
+    return (
+      <sc-dial-tooltip
+        title={title(propertyStream)}
+        propertyPoint={this.propertyPoint}
+        alarmPoint={this.alarmStream && this.propertyPoint}
+        breachedThreshold={this.breachedThreshold}
+        unit={unit || '%'}
```

Comment:
should be `unit ?? '%'` or `unit != null ? unit : '%'', otherwise if someone puts unit as '' theyll get a percentage sign

Reply:
`unit || '%'` means that  if 'unit === null' or 'unit === undefined' or 'unit === '' ' ,that will get '%' , doesn't it?

Reply:
that is incorrect - try it for yourself. please do not resolve comments that shouldn't be resolved